misc/custom_promise_types/cfengine.py

```diff
@@ -291,7 +291,15 @@ def _handle_validate(self, promiser, attributes):
                 # Bad, validate method shouldn't return anything else
                 self._result = Result.ERROR
         except ValidationError as e:
-            self.log_error(e)
+            message = str(e)
+            if "promise_type" in request:
+                message += f" for {request['promise_type']} promise with promiser '{promiser}'"
+            else:
+                message += f" for promise with promiser '{promiser}'"
+            if "filename" in request and "line_number" in request:
+                message += f" ({request['filename']}:{request['line_number']})"
```

Comment:
There must be a missing `"\n"` somewhere in the above lines.

Reply:
No, it is correct, the log message looks like this:

```
error: Unknown attribute 'versionss' for git promise with promiser 'starter_pack_repo' (example.cf:11)
```